cpp/tensorrt_llm/batch_manager/cacheTransBuffer.cpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cacheTransBuffer.h"
+#include "tensorrt_llm/common/envUtils.h"
+#include "tensorrt_llm/common/logger.h"
+#include <NvInferRuntimeBase.h>
+#include <mutex>
+
+namespace tensorrt_llm::batch_manager::kv_cache_manager
+{
+
+CacheTransBufferManager::CacheTransBufferManager(
+    KVCacheManager::BaseKVCacheManager* cacheManager, std::optional<size_t> maxNumTokens)
+    : mCacheManager{cacheManager}
+    , mBufferManager{std::make_shared<runtime::CudaStream>()}
+{
+
+    TLLM_CHECK(mCacheManager);
+    mDataType = mCacheManager->getPrimaryPool(0)->getDataType();
+
+    auto tokensPerBlock = mCacheManager->getBlockManager().getTokensPerBlock();
+    if (maxNumTokens.has_value())
+    {
+        TLLM_CHECK(maxNumTokens.value() % tokensPerBlock == 0);
+    }
+    TLLM_LOG_INFO("maxNumTokens: %d", maxNumTokens.has_value() ? maxNumTokens.value() : 0);
+    auto kvCachePerToken
+        = (mCacheManager->getBlockManager().getBlockSize(0) * mCacheManager->getBlockManager().getNumLayers()
+              * (mCacheManager->getCacheType() == CacheType::kSELFKONLY ? 1 : 2))
+        / tokensPerBlock;
+    mTransferBufferSize = maxNumTokens.has_value() ? maxNumTokens.value() * kvCachePerToken
+                                                   : common::getEnvMemSizeForKVCacheTransferBuffer();
+    monlyUseDynamicBuffer = mTransferBufferSize == 0;
+    mRecvBufferCount = common::getEnvRequestKVCacheConcurrent() ? common::getEnvKVCacheRecvBufferCount() : 1;
```

Comment:
Currently, we have a configuration data structure. Could we move all the environment variables involved to `CacheTransceiverConfig` in the next PRs?